### DISCUSSION TITLE
Terrible loading time on mobile devices (espesially on poor performance devices)

### DISCUSSION THREAD

**Discussion by kopeek** [godotengine/godot-proposals#11402] [Platforms]
### Describe the project you are working on

I'm working on a 3D platformer Android game on smartwatch.

### Describe the problem or limitation you are having in your project

I have a big problem with these games and apps made in Godot. their loading time is insane. my 3D game loads in around 17 seconds.
I tried to make a blank app with text that changes, and it is still taking around 9 seconds to load. My Smartwatch isn't so bad. It can run that 3D game in around 46 fps (not accurate). and it has around 1,5 GB or RAM. And And it's worth noting that this nearly empty app took 9 seconds to load. Why it is taking so long, for a blank app to open?

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Optimize the export, so the app can load faster (3 seconds would be good), and make better use of poor performance devices.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

It can be in the options in export, but maybe a custom build.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

I can't even make a normal app in Godot for those devices, because the time for loading the app is too long, users probably don't want to wait 10 seconds for a simple app to load

**Comment by AThousandShips** [comment]
This is a bug not a missing feature, please make a bug report [here](https://github.com/godotengine/godot/issues) instead

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

> I'm working on a 3D platformer Android game on smartwatch.

From a developer standpoint, smartwatches are difficult to optimize for. They have *extremely* limited processing power compared to smartphones. To give a frame of reference, [Apple Watches can just about run Quake 2 as it is](https://github.com/ByteOverlord/Watch_Quake2) while phones have been able to do that for roughly 15 years now.

**Comment by Zireael07** [comment]
@Calinou Why the assumption this is a web export? I see nothing that would indicate this

**Comment by kopeek** [comment]
> Thanks for the report! Consolidating in [godotengine/godot#41118](https://github.com/godotengine/godot/issues/41118).
> 
> > I'm working on a 3D platformer Android game on smartwatch.
> 
> From a developer standpoint, Web exports on mobile platforms are difficult to optimize for. This gets worse when you're targeting smartwatches which have _extremely_ limited processing power compared to smartphones. To give a frame of reference, [Apple Watches can just about run Quake 2 as it is](https://github.com/ByteOverlord/Watch_Quake2) while phones have been able to do that for roughly 15 years now.
> 
> Regarding startup time performance, _native_ 3D games already struggle to open in 3 seconds on a smartwatch.

Please read my problem again carefully. I wasn't using web export, normal Android export, and I have also said that even app not games with barelly anything in them, can load in 9 seconds
